ksql-rest-app/src/main/java/io/confluent/ksql/rest/server/computation/CommandRunner.java

```diff
@@ -64,11 +64,13 @@
 
   private final CommandRunnerStatusMetric commandRunnerStatusMetric;
   private final AtomicReference<Pair<QueuedCommand, Instant>> currentCommandRef;
+  private final AtomicReference<Instant> lastPollTime;
   private final Duration commandRunnerHealthTimeout;
   private final Clock clock;
 
   public enum CommandRunnerStatus {
     RUNNING,
+    STUCK,
```

Comment:
why do we need this additional status? Once it's here we have to support it forever. Why is it not enough to know that the runner is up or down?

Reply:
I thought it would be helpful to be able to tell the difference between if the CommandRunner thread is stuck on a particular Command or the thread itself has died.

If we do force the server to shut down whenever the CommandRunner thread dies, then this additional status won't be needed.